### PR TITLE
Supervised learning datasets

### PR DESCRIPTION
**Pull request by zhangguanheng66**
Add a few supervised learning datasets.
Add a tutorial to use the new dataset for sentiment analysis.

### CODE REVIEW

**Comment by zhangguanheng66** [comment]
@facebook-github-bot retest this please

File: docs/tutorials/text_sentiment_ngrams.ipynb [hunk 1]
```diff
@@ -0,0 +1,375 @@
+{
```
Comment (cpuhrsch) [inline docs/tutorials/text_sentiment_ngrams.ipynb]: Can you split this out into a separate PR just to make this one smaller?

Comment (zhangguanheng66) [reply docs/tutorials/text_sentiment_ngrams.ipynb]: NP

File: test/data/test_builtin_datasets.py [hunk 2]
```diff
@@ -47,3 +49,25 @@ def test_penntreebank(self):
...
+        txt_cls = TextClassificationDataset("ag_news", root=datadir, ngrams=2)
+        train_iter, test_iter, valid_iter = txt_cls.iters(batch_size=128,
+                                                          device="cpu")
+        ag_news_cls = AG_NEWS(root=datadir, ngrams=3)
+        train_iter, test_iter, valid_iter = ag_news_cls.iters(batch_size=128,
+                                                              device="cpu")
+
+        # Delete the dataset after we're done to save disk space on CI
+        if os.environ.get("TRAVIS") == "true":
+            datafile = os.path.join(self.project_root,
```
Comment (cpuhrsch) [inline test/data/test_builtin_datasets.py]: Instead of this you should be able to create a temporary directory via https://docs.python.org/3/library/tempfile.html , which has all of this built in and is therefore safer.

File: torchtext/datasets/text_classification.py [hunk 3]
```diff
@@ -0,0 +1,484 @@
...
+import re
+import torch
+from ..utils import download_and_extract_archive, makedir_exist_ok
+import random
+from .. import data
+from tqdm import tqdm
+from ..data import dataset
+
+
+class TextClassificationDataset(data.Dataset):
```
Comment (cpuhrsch) [inline torchtext/datasets/text_classification.py]: Is it possible to avoid creating this and have each dataset (e.g. AG_NEWS ) use only https://pytorch.org/docs/stable/data.html instead?

torchtext dataset abstractions need a lot of work and we should avoid the proliferation of torchtext-style datasets until we did a wider review.

This should be fine even if it means that we first release AG_NEWS, etc. with less features.
